syntax = "proto3";

option java_multiple_files = true;
option java_package = "grpc.proto";
option java_outer_classname = "CVProto";

service CV {
    rpc Subscription (stream Request) returns (stream CurriculumVitae);
}

message Request {
    enum Action {
        START = 0;
        STOP = 1;
    }
    Action action = 1;
    Country country = 2;
}


message CurriculumVitae {
    Person person = 1;
    repeated Job previousJobs = 2;
    string description = 3;
}

message Job {
    string job = 1;
    string company = 2;
}

message Person {
    string firstName = 1;
    string secondName = 2;
    Birth birth = 3;
    string email = 4;
    string phoneNumber = 5;
}

message Birth {
    Country country = 1;
    string date = 2;
}

enum Country {
    US = 0;
    Poland = 1;
    Germany = 2;
    France = 3;
    Japan = 4;
}
